external/fv3fit/fv3fit/keras/_models/precipitative.py

```diff
@@ -35,27 +35,21 @@
 Q_TENDENCY_NAME = "dQ2"
 
 
-def integrate_precip(args):
-    dQ2, delp = args  # layers seem to take in a single argument
-    # output should be kg/m^2/s
-    return tf.math.scalar_mul(
-        # dQ2 * delp = s-1 * Pa = s^-1 * kg m-1 s-2 = kg m-1 s-3
-        # dQ2 * delp / g = kg m-1 s-3 / (m s-2) = kg/m^2/s
-        tf.constant(-1.0 / GRAVITY, dtype=dQ2.dtype),
-        tf.math.reduce_sum(tf.math.multiply(dQ2, delp), axis=-1),
-    )
-
+class IntegratePrecipLayer(tf.keras.layers.Layer):
+    def call(self, args) -> tf.Tensor:
+        dQ2, delp = args  # layers seem to take in a single argument
+        # output should be kg/m^2/s
+        return tf.math.scalar_mul(
+            # dQ2 * delp = s-1 * Pa = s^-1 * kg m-1 s-2 = kg m-1 s-3
+            # dQ2 * delp / g = kg m-1 s-3 / (m s-2) = kg/m^2/s
+            tf.constant(-1.0 / GRAVITY, dtype=dQ2.dtype),
+            tf.math.reduce_sum(tf.math.multiply(dQ2, delp), axis=-1),
+        )
 
-def condensational_heating(dQ2):
-    """
-    Args:
-        dQ2: rate of change in moisture in kg/kg/s, negative corresponds
-            to condensation
 
-    Returns:
-        heating rate in degK/s
-    """
-    return tf.math.scalar_mul(tf.constant(-LV / CPD, dtype=dQ2.dtype), dQ2)
+class CondensationalHeatingLayer(tf.keras.layers.Layer):
```

Comment:
I would keep the documentation describing the specifics of the values used/output by this layer.

Reply:
Restored, thanks.